handlers/challenge_handler.py

```diff
@@ -410,6 +410,68 @@ def execute(self, slack_wrapper, args, channel_id, user_id):
         # Show confirmation message
         slack_wrapper.post_message(channel_id, message)
 
+class AddCredsCommand(Command):
+    """Add credential informations for current ctf"""
+
+    def execute(self, slack_wrapper, args, channel_id, user_id):
+        """Execute the AddCreds command."""
+
+        cur_ctf = get_ctf_by_channel_id(ChallengeHandler.DB, channel_id)
+        if not cur_ctf:
+            raise InvalidCommand(
+                "Command failed. You are not in a CTF channel.")
+
+        # Update database
+        ctfs = pickle.load(open(ChallengeHandler.DB, "rb"))
+
+        for ctf in ctfs.values():
+            if ctf.name == cur_ctf.name:
+                ctf.cred_user = args[0]
+                ctf.cred_pw = args[1]
+                ctf.cred_url = args[2] if len(args) > 2 else ""
+
+                pickle.dump(ctfs, open(ChallengeHandler.DB, "wb"))
+
+                purpose = dict(ChallengeHandler.CTF_PURPOSE)
+                purpose["ota_bot"] = "DO_NOT_DELETE_THIS"
+                purpose["name"] = ctf.name
+                purpose["type"] = "CTF"
+                purpose["cred_user"] = ctf.cred_user
+                purpose["cred_pw"] = ctf.cred_pw
+                purpose["cred_url"] = ctf.cred_url
+
+                slack_wrapper.set_purpose(ctf.channel_id, purpose)
+
+                message = "Credentials for CTF *{0}* updated...".format(ctf.name)
+                slack_wrapper.post_message(channel_id, message)
+                return
+    
+class ShowCredsCommand(Command):
+    """Shows credential informations for current ctf"""
+
+    def execute(self, slack_wrapper, args, channel_id, user_id):
+        """Execute the ShowCreds command."""
+
+        cur_ctf = get_ctf_by_channel_id(ChallengeHandler.DB, channel_id)
+        if not cur_ctf:
+            raise InvalidCommand(
+                "Command failed. You are not in a CTF channel.")
+
+        if (cur_ctf.cred_user != "") or (cur_ctf.cred_pw != ""):
+            message = "Credentials for CTF *{0}*\n".format(cur_ctf.name)
+            message += "```"
+
+            if (cur_ctf.cred_url != ""):
```

Comment:
Maybe `if cur_ctf.cred_url:` ?

Additional comment : You don't need to wrap if statements in parenthesis :)

Reply:
yep, that's just a bad habbit coming from another language ;)